client/src/components/graph/graph.tsx

```diff
@@ -614,17 +604,16 @@ class Graph extends React.Component<GraphProps, GraphState> {
     return { toolSVG: newToolSVG, tool, container };
   };
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  loadTextureFromUrl = async (src: string): Promise<any> => {
-    this.downloadedImg.crossOrigin = "anonymous";
-    this.downloadedImg.src = src;
+  loadTextureFromProp = async (src: string): Promise<HTMLImageElement> => {
+    this.underlayImage.crossOrigin = "anonymous";
+    this.underlayImage.src = src;
 
     await new Promise((resolve, reject) => {
-      this.downloadedImg.onload = () => resolve(this.downloadedImg);
-      this.downloadedImg.onerror = reject;
+      this.underlayImage.onload = () => resolve(this.underlayImage);
```

Comment:
I think I'm confused on the structure of this promise that we're creating. What exactly in here is asynchronus?

Reply:
True - this is leftover from when we had a dedicated endpoint for this